src/handlers.jl

```diff
@@ -49,6 +49,51 @@ Base.chr2ind(m::Msg, str::String, ic::Integer) = ic == 0 ? 0 :
     VersionNumber(m.header["version"]) ≥ v"5.2" ? chr2ind(str, ic) : utf16_to_ind(str, ic)
 Base.ind2chr(m::Msg, str::String, i::Integer) = i == 0 ? 0 :
     VersionNumber(m.header["version"]) ≥ v"5.2" ? ind2chr(str, i) : ind_to_utf16(str, i)
+#Compact display of types for Jupyterlab completion
+
+complete_type(::Type{<:Function}) = "function"
+complete_type(::Type{<:Type}) = "type"
+complete_type(::Type{<:Tuple}) = "tuple"
+
+function complete_type(T::DataType)
+    s = string(T)
+    (Compat.textwidth(s) ≤ 20 || isempty(T.parameters)) && return s
+    buf = IOBuffer()
+    print(buf, T.name)
+    position(buf) > 19 && return String(take!(buf))
+    print(buf, '{')
+    comma = false
+    for p in T.parameters
+        s = string(p)
+        if position(buf) + sizeof(s) > 20
+            comma || print(buf, '…')
+            break
+        end
+        comma && print(buf, ',')
+        comma = true
+        print(buf, s)
+    end
+    print(buf, '}')
+    return String(take!(buf))
+end
+
+#Get typeMap for Jupyter completions
+function complete_types(comps)
+    typeMap = []
+    for c in comps
+        expr = Meta.parse(c, raise=false)
+        if typeof(expr) == Symbol
+            try
+                t = eval(Main, Meta.parse("typeof($c)"))
+                push!(typeMap, Dict("text" => c, "type" => "$(complete_type(t))"))
+            catch
+            end
+        elseif expr.head == :macrocall
```

Comment:
Note that this assumes `isa(expr, Expr)`, but this is now correct with my suggested change above.